internal/services/resource/resource_provider_registration_resource.go

```diff
@@ -49,6 +50,11 @@ const (
 	Unregistered  = "Unregistered"
 )
 
+func (r ResourceProviderRegistrationResource) DeprecationMessage() string {
+	// TODO: Remove this resource in v4.0
```

Comment:
If we're deprecating and removing this we should also put the registration of this resource behind the 4.0 flag

Reply:
Actually we decided on keeping this resource for the explicit functionality it provides - I committed this in error. Will fix 👍